src/Twig/Component/Banner.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace EasyCorp\Bundle\EasyAdminBundle\Twig\Component;
+
+use EasyCorp\Bundle\EasyAdminBundle\Config\Enum\BannerVariant;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\UX\TwigComponent\Attribute\PreMount;
+
+class Banner
+{
+    public BannerVariant|string $variant = BannerVariant::Info;
+    public bool $withDismissButton = false;
+
+    #[PreMount]
+    public function preMount(array $data): array
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setIgnoreUndefined(true);
+
+        $resolver->define('variant')
+            ->allowedTypes('string', BannerVariant::class)
+            ->allowedValues(...array_map(static fn(BannerVariant $variant): string => $variant->value, BannerVariant::cases()));
+
+        $resolver->define('withDismissButton')
+            ->allowedTypes('bool');
+
+        return $resolver->resolve($data) + $data;
+    }
+
+    public function getVariantCssClass(): string
+    {
+        return \is_string($this->variant) ? $this->variant : $this->variant->asCssClass();
+    }
```

Comment:
```suggestion

    public BannerVariant $variant = BannerVariant::Info;

    public function mount(string|BannerVariant|null $variant): void
    {
           if (is_string($variant)) {
             $variant = BannerVariant::tryFrom($variant) ?? throw new Invalid...
           }
        
           $this->variant = $variant ?? $this->variant;
    }

    public function getVariantCssClass(): string
    {
        return $this->variant->asCssClass();
    }
```

Maybe simpler to maintain.

Reply:
+1 

It can even be simplified to `$this->variant = is_string($variant) ? BannerVariant::from($variant) : $variant`.

Also, what it this `mount` method? Something I don't know yet? 👀

Reply:
```
$this->variant = is_string($variant) ? BannerVariant::from($variant)
```

Did not do it because it make the validation on render and not on mount... but on a pragmatic level it's the same :)